static/app/views/performance/queues/destinationSummary/transactionsTable.tsx

```diff
@@ -0,0 +1,213 @@
+import {browserHistory, Link} from 'react-router';
+import styled from '@emotion/styled';
+import type {Location} from 'history';
+import qs from 'qs';
+
+import GridEditable, {
+  COL_WIDTH_UNDEFINED,
+  type GridColumnHeader,
+} from 'sentry/components/gridEditable';
+import type {CursorHandler} from 'sentry/components/pagination';
+import Pagination from 'sentry/components/pagination';
+import {t} from 'sentry/locale';
+import type {Organization} from 'sentry/types';
+import type {EventsMetaType} from 'sentry/utils/discover/eventView';
+import {getFieldRenderer} from 'sentry/utils/discover/fieldRenderers';
+import {decodeScalar} from 'sentry/utils/queryString';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
+import {renderHeadCell} from 'sentry/views/starfish/components/tableCells/renderHeadCell';
+import {useSpanMetrics} from 'sentry/views/starfish/queries/useSpanMetrics';
+import type {MetricsResponse} from 'sentry/views/starfish/types';
+import {QueryParameterNames} from 'sentry/views/starfish/views/queryParameters';
+
+type Row = Pick<
+  MetricsResponse,
+  | 'avg_if(span.self_time,span.op,queue.task.celery)'
+  | 'count_op(queue.submit.celery)'
+  | 'count_op(queue.task.celery)'
+  | 'sum(span.self_time)'
+  | 'transaction'
+>;
+
+type Column = GridColumnHeader<string>;
+
+const COLUMN_ORDER: Column[] = [
+  {
+    key: 'transaction',
+    name: t('Transactions'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: '', // TODO
+    name: t('Type'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: '', // TODO
+    name: t('Avg Time in Queue'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'avg_if(span.self_time,span.op,queue.task.celery)',
+    name: t('Avg Processing Time'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: '', // TODO
+    name: t('Error Rate'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'count_op(queue.submit.celery)',
+    name: t('Published'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'count_op(queue.task.celery)',
+    name: t('Processed'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'sum(span.self_time)',
+    name: t('Time Spent'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+];
+
+interface Props {
+  domain?: string;
+  error?: Error | null;
+  meta?: EventsMetaType;
+  pageLinks?: string;
+}
+
+export function TransactionsTable({error, pageLinks}: Props) {
+  const organization = useOrganization();
+  const location = useLocation();
+  const destination = decodeScalar(location.query.destination);
+  const cursor = decodeScalar(location.query?.[QueryParameterNames.DOMAINS_CURSOR]);
+
+  const mutableSearch = new MutableSearch(
+    'span.op:[queue.task.celery,queue.submit.celery]'
+  );
+  if (destination) {
+    mutableSearch.addFilterValue('transaction', destination);
+  }
```

Comment:
IMO it would be a good idea to use the name "transaction" everywhere except the user-facing label. If the UI label for a transaction is "destination" that's  totally fine, but if under the hood destinations are just transactions, the code should talk about transactions! Otherwise it's confusing

Reply:
So this filter should actually be filtering by a `destination` tag, not transaction. Since `destination` isn't yet tagged or implemented in the sdks, I'm using transaction temporarily to continue building out the product UI. Added a TODO comment around this to be more clear about this situation!

Reply:
Ah, okay!